src/components/adminDashboard/components/eMetadata/services/eMetadata.service.js

```diff
@@ -0,0 +1,121 @@
+             //   var profilePath = "http://localhost:3009/teachworks/zinkerzTeacher/all";
+                var globalStorageProm = InfraConfigSrv.getGlobalStorage();
+
+                var satURL = "https://sat-dev.firebaseio.com";
+                var actURL = "https://act-dev.firebaseio.com";
+                var tofelURL = "https://znk-toefl-dev.firebaseio.com";
+
+                if (!ENV.debug) {
+                    // satURL = "https://sat-dev.firebaseio.com/";
+                    // actURL = "https://act-dev.firebaseio.com/";
+                    // tofelURL = "https://znk-toefl-dev.firebaseio.com/";
+                }
+
+
+                self.showEducatorProfile = function (userProfile) {
+                    $q.all([getTimezonesList(), getLocalTimezone()]).then(function (values) {
+                        var timezonesList = values[0];
+                        var localTimezone = values[1];
+                        $mdDialog.show({
+                            locals: {
+                                userProfile: userProfile,
+                                timezonesList: obj2Array(timezonesList),
+                                localTimezone: localTimezone
+                            },
+                            controller: 'EducatorProfileController',
+                            controllerAs: 'vm',
+                            templateUrl: 'components/adminDashboard/components/eMetadata/templates/educatorProfile.template.html',
+                            clickOutsideToClose: true,
+                            escapeToClose: true
+                        });
+                    });
+
+                };
+
+                self.updateProfile = function (newProfile) {
+                    var copiedProfile =angular.copy(newProfile);
+                    var uid = copiedProfile.uid;
+                    if (uid) {
+                        delete copiedProfile.uid;
+                    }
+                    var fullPath = "users/" + uid + "/profile";
+                    return InfraConfigSrv.getGlobalStorage().then(function (globalStorage) {
+                        return globalStorage.update(fullPath, copiedProfile);
+                    });
+                };
+                self.setZinkerzTeacher = function (uid, subject, isZinkerzTeacher) {
+                    if (!uid) {
+                        $log.error('setZinkerzTeacher: no uid');
+                        return;
+                    }
+                    if (!subject) {
+                        $log.error('setZinkerzTeacher: no subject');
+                        return;
+                    }
+                    var profile = {
+                        userId: uid,
+                        isZinkerzTeacher: !!isZinkerzTeacher,
+                        teachingSubject: subject,
+                        fbUrls: [satURL, actURL, tofelURL]
+                    };
+                    return $http.post(profilePath, profile);
```

Comment:
what will be if there is an error? 

You can also look on timeout promise we have